addon/components/inputs/checkbox-array.js

```diff
@@ -27,13 +35,36 @@ export default AbstractInput.extend({
   },
 
   @readOnly
-  @computed('bunsenModel')
-  options (bunsenModel) {
+  @computed('bunsenModel', 'cellConfig')
+  options (bunsenModel, cellConfig) {
     const items = get(bunsenModel, 'items.enum') || []
+    const choices = _.get(cellConfig, 'renderer.choices') || []
+    const meta = _.get(cellConfig, 'renderer.meta') || []
+
+    helpers.validateChoices(choices, meta)
 
-    return items.map((value) => {
-      return { value }
-    })
+    // Initialize selected for each model property
+    this.get('selected')[this.get('bunsenId')] = []
```

Comment:
This change is the wrong approach. You have a `selected` property set on the class itself which is being shared across all instances. The proper way to do this would instead be:

```js
const assign = Object.assign || Ember.assign || Ember.merge
…
  getDefaultProps () {
    const defaults = this._super(...arguments)
    return assign(defaults, {
      selected: []
    })
  }
```

This will give each instance it's own value for `selected` and you can maintain the array approach.